### PR TITLE
fix: open house events can now be edited and work cross-browser

### PR DESCRIPTION
**Pull request by jaredcwhite**
## Issue Overview

This PR addresses #2265 and #2263 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Fixes for both issues mentioned above: previously an existing open house event wouldn't get saved properly if you tried to edit it. In addition, open house dates (and other dates like application due date/time) would result in "Invalid Date" after adding/editing an open house in browsers like Firefox and Safari.

## How Can This Be Tested/Reviewed?

Ensure open house events can be added/edited/saved without error or data loss in Chromium-based browsers, Firefox, and Safari.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/partners/src/listings/PaperListingDetails/sections/DetailApplicationDates.tsx [hunk 1]
```diff
@@ -32,6 +32,11 @@ const DetailApplicationDates = () => {
     () =>
       listing.events
         .filter((item) => item.type === ListingEventType.openHouse)
+        .sort((a, b) => {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          return new Date(a.startTime) - new Date(b.startTime)
+        })
```
Comment (dominikx96) [inline sites/partners/src/listings/PaperListingDetails/sections/DetailApplicationDates.tsx:39]: I think you don't need to ignore TS warning here, a `getTime()` method might help to get numeric value, like:

```
return new Date(a.startTime).getTime() - new Date(b.startTime).getTime()
```

so now that's the arithmetic operation.

**Review by dominikx96 — CHANGES_REQUESTED** [review]
@jaredcwhite works good! I've just put one comment about ignoring TS warnings.

File: sites/partners/lib/helpers.ts [hunk 2]
```diff
@@ -162,7 +162,18 @@ export const createTime = (
...
+
+  if (tzOffset >= 0) {
+    if (tzOffset < 10) tzString = `+0${tzOffset}:00`
+    if (tzOffset >= 10) tzString = `+${tzOffset}:00`
+  } else {
+    if (tzOffset > -10) tzString = `-0${tzOffset.toString().substring(1)}:00`
+    if (tzOffset <= -10) tzString = `${tzOffset}:00`
+  }
+
+  return new Date(`${formDate.year}-${formDate.month}-${formDate.day}T00:00:00.000${tzString}`)
```
Comment (seanmalbert) [inline sites/partners/lib/helpers.ts:176]: Can we use dayjs to handle some of this, like setting the utc offset. I know adding dayjs is in another PR, but using that could probably take some logic out of our hands.

**Review by seanmalbert — CHANGES_REQUESTED** [review]

File: sites/partners/lib/helpers.ts (hunk 2 above)
Comment (jaredcwhite) [reply sites/partners/lib/helpers.ts:176]: Agreed, I worked on this before working on the other brach. Up to you if you want to merge this as-is or redo after adding day.js

Comment (jaredcwhite) [reply sites/partners/lib/helpers.ts:172]: @seanmalbert All right, the dayjs stuff is merged in now, so I'll work on updating this and getting rid of so much custom logic.

**Comment by jaredcwhite** [comment]
I've pushed updates up to utilize dayjs more in this PR, including in sort functions.

**Review by seanmalbert — APPROVED** [review]
Thanks for the updates, @jaredcwhite. I updated `getDetailFieldTime` to not include the seconds and made an unrelated change to the form's useEffect hook dependencies, since that was causing issues.

**Pull request merged**